### PR TITLE
feat(docs): update micro-frontends deployment guide, versioning, tags, and more

### PR DESCRIPTION
**Pull request by zmzlois**
### What's added in this PR?

- Add supported cloud list and what's under development 
- Component: Badge, Code, Copy Button, Invert image, progress bar, smooth-pre, Tooltip, Tabs
- Display latest npm version at the top of the doc
- Custom console output for a standard Zephyr deployment
- Fix `Create MF App` guide 
- Fix deadlinks
- Fix FAQ
- Tidy up browser extension usage guide 
- Clean up Cloud provider guide and add `Clean uninstall` 
- Add "Deploy MF with Zephyr complete guide"
- Tidy up tags and versioning usage in dashboard
- Update assets for Architecture guide 
- Learning section init 
- Add fonts 

### What's the issues or discussion related to this PR (optional) ?

N/A

### Feature related update for this PR (if applicable)?
- Environment and tags

### (Optional) What's left to be done for this PR?
- Add open graph image to docs
- Fix Nx related issues on Recipes
- walk through, clean up examples to pair with examples repo
- Add dream-mf as an example (for when users are not using Module Federation what should they do)
- Reverify Architecture guide 
- Add learning section for Terminologies, Module Federation, Micro-Frontends and Edge 

### Who do you wish to review this PR other than required reviewers?
@zackarychapple @valorkin 

### (Required) Pre-PR/Merge checklist

- [ ] I have added/updated our feature to sync with this PR
- [x] I have added an explanation of my changes
- [ ] I have/will run tests, or ask for help to add test

### CODE REVIEW

File: docs/how-to/mf-guide.mdx [hunk 1]
```diff
@@ -0,0 +1,165 @@
+import { Steps, Badge } from '@theme';
+import { Separator } from "../../components/ui/separator"
+
+# Micro-Frontends with Zephyr - The Complete Guide
+
+Micro-Frontends' configuration and it's deployment processes are known to be complicated and painful to debug. Zephyr came in not only to help with resolving dynamic URL and development process for applications applying Micro-Frontends architecture, but also deployment, versioning and more. This section will walk through what are the needed configuration to ensure you have a smooth development workflow and deployment process.
```
Comment (zackarychapple) [inline docs/how-to/mf-guide.mdx:6]:
```suggestion
Micro-Frontends' configuration and deployment processes are known to be complicated and painful to debug. Zephyr Cloud not only to helps with resolving dynamic URL and development process for applications applying Micro-Frontends architecture, but also deployment, versioning and more. This section will walk through the necessary configuration to ensure you have a smooth development workflow and deployment process.
```

File: docs/how-to/mf-guide.mdx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+Micro-Frontends' configuration and it's deployment processes are known to be complicated and painful to debug. Zephyr came in not only to help with resolving dynamic URL and development process for applications applying Micro-Frontends architecture, but also deployment, versioning and more. This section will walk through what are the needed configuration to ensure you have a smooth development workflow and deployment process. 
+
+## Checklist
+
+
+<Steps>
+### Micro-Frontends Configuration 
+
+Our first class supported Micro-Frontends architecture is [Module Federation](https://module-federation.io), while other Micro-Frontends architecture might be supported.
```
Comment (zackarychapple) [inline docs/how-to/mf-guide.mdx:14]:
```suggestion
We have first class support for a Micro-Frontends architecture based on [Module Federation](https://module-federation.io). Other Micro-Frontends architectures may already work or will be supported in the future. 
```

File: docs/how-to/mf-guide.mdx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+### Build remotes first 
+
+At the moment. when we are resolving your remote dependencies, we will need the [remotes](/learning/concepts/terminologies#remote-application) to be built first. Be wary that the first built remotes should not be consuming any other remotes. 
+
+In the example below where you have one shell and one remote, you will need to build the remote first. 
+
+![mf_1_circular_dependencies](/mf_guide_1_circular_dependency.png)
+
+Let's take a look at another example. You have a shell applicaiton, within the shell app you have two modules: **HeaderNav** and **TaskContent**.
```
Comment (zackarychapple) [inline docs/how-to/mf-guide.mdx:59]:
```suggestion
Let's take a look at another example. You have a shell application, within the shell app you have two modules: **HeaderNav** and **TaskContent**. 
```

File: docs/how-to/mf-guide.mdx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+![mf_2_circular_dependencies](/mf_guide_2_circular_dependency.png)
+
+The **HeaderNav** exposes **Logo**, **UserProfileInfo** and **LeftNav**. 
+
+The **TaskContent** exposes **SearchBar**. 
+
+In this scenario, **LeftNav** is consumed by **TaskContent**; **SearchBar** is consumed by **HeaderNav**. 
+
+You will need to build either the **TaskContent** or the **HeaderNav** first, but comment out the remotes that have yet build and add them back later.
```
Comment (zackarychapple) [inline docs/how-to/mf-guide.mdx:69]:
```suggestion
You will need to build either the **TaskContent** or the **HeaderNav** first, comment out the remotes that have yet built and add them back later. 
```

**Review by zackarychapple — CHANGES_REQUESTED** [review]

File: docs/how-to/mf-guide.mdx (hunk 1 above)
Comment (zmzlois) [reply docs/how-to/mf-guide.mdx]: https://github.com/ZephyrCloudIO/zephyr-documentation/pull/12/commits/369b71f277deffafcfd603fa2225db71c8388dc2

File: docs/how-to/mf-guide.mdx (hunk 3 above)
Comment (zmzlois) [reply docs/how-to/mf-guide.mdx:59]: https://github.com/ZephyrCloudIO/zephyr-documentation/pull/12/commits/cc9fc3d376ebabe752e233c675d250de8b067da1

File: docs/how-to/mf-guide.mdx (hunk 2 above)
Comment (zmzlois) [reply docs/how-to/mf-guide.mdx]: https://github.com/ZephyrCloudIO/zephyr-documentation/pull/12/commits/cc9fc3d376ebabe752e233c675d250de8b067da1

File: docs/how-to/mf-guide.mdx (hunk 4 above)
Comment (zmzlois) [reply docs/how-to/mf-guide.mdx]: https://github.com/ZephyrCloudIO/zephyr-documentation/pull/12/commits/a1627a35c5b5c8212c29bf29937c49730c71ba0a

**Pull request merged**